### PR TITLE
Uni 160 display uploaded file metrics in left menu

### PR DESCRIPTION
**Pull request by marionleborgne**
WIP

### CODE REVIEW

File: unicorn/frontend/lib/FileServer.js [hunk 1]
```diff
@@ -86,6 +86,36 @@ FileServer.prototype.getSampleFiles = function(callback) {
...
+    var formattedFile = {
+        name: file.name,
+        filename: file.path,
+        type: 'uploaded',
+        metrics: []
+      };
+      callback(formattedFile);
+
+    /*
+    @TODO trying to popuplate metrics with getFields
```
Comment (marionleborgne) [inline unicorn/frontend/lib/FileServer.js]: @brev @lscheinkman I managed to have some info of the uploaded file to display in the left menu (file name), but I couldn't get the metrics name to display. I'm trying to populate  formattedFile.metrics with the metric names via getFields, but for some reason that method never gets called. I commented it out for now, and placed the callback above it.

Any idea how I could get this list of metrics names? Maybe that is not the right spot to do that?

Comment (marionleborgne) [inline unicorn/frontend/lib/FileServer.js]: Thanks @lscheinkman it works now

**Comment by marionleborgne** [comment]
@lscheinkman @brev ready for review :)

**Comment by brev** [comment]
:+1: this is looking great @marionleborgne @lscheinkman thanks :)

File: unicorn/frontend/lib/FileServer.js [hunk 2]
```diff
@@ -101,7 +127,9 @@ FileServer.prototype.getSampleFiles = function(callback) {
  *                             }
  *                             </code>
  */
-FileServer.prototype.getFields = function(filename, options, callback) {
+ FileServer.prototype.getFields = function(filename, options, callback) {
+
+  debugger;
```
Comment (lscheinkman) [inline unicorn/frontend/lib/FileServer.js]: @marionleborgne Please remove breakpoints and other debugging artifacts before merging

File: unicorn/frontend/browser/components/FileList.jsx [hunk 3]
```diff
@@ -68,13 +68,16 @@ class FileListComponent extends React.Component {
...
   }
-  _getFileItems() {
+  _getFileItems(filetype) {
     return this.props.files.map(file => {
-      return (
-        <ListItem initiallyOpen={true}
-          key={file.name}
-          nestedItems={this._getMetricItems(file)} primaryText={file.name}/>
-      );
+      console.log("FILE: ", file);
```
Comment (lscheinkman) [inline unicorn/frontend/browser/components/FileList.jsx]: @marionleborgne Are you sure we need all these extra `console.log` calls?

File: unicorn/frontend/browser/components/Main.jsx [hunk 4]
```diff
@@ -71,40 +72,43 @@ module.exports = React.createClass({
...
+
+    var file = files[0];
+
+    console.log("Selected file: ", files);
+
+    console.log('got clicked! firing FileUploadAction.');
+    
+    this.context.executeAction(FileUploadAction, file);
+    console.log('FileUploadAction should have fired.');
+
```
Comment (lscheinkman) [inline unicorn/frontend/browser/components/Main.jsx]: @marionleborgne Are you sure we need all these extra `console.log` calls?

File: unicorn/frontend/lib/FileServer.js [hunk 5]
```diff
@@ -86,6 +86,32 @@ FileServer.prototype.getSampleFiles = function(callback) {
...
+  var formattedFile = {
+    name: file.name,
+    filename: file.path,
+    type: 'uploaded',
+    metrics: []
+  };
+
+  this.getFields(formattedFile.filename, {} ,(error, fields) => {
+    if (error) {
+      console.log('Error loading metrics for file:', formattedFile, error);
```
Comment (lscheinkman) [inline unicorn/frontend/lib/FileServer.js]: The first parameter of the `callback` is the error.
In addition to logging the error, you should notify the caller of the error, something like this:
```javascript
callback(error);
```

File: unicorn/frontend/browser/components/Main.jsx (hunk 4 above)
Comment (marionleborgne) [inline unicorn/frontend/browser/components/Main.jsx]: no we don't. I was trying to debug my initial error :innocent:

File: unicorn/frontend/lib/FileServer.js (hunk 5 above)
Comment (marionleborgne) [inline unicorn/frontend/lib/FileServer.js]: added

**Comment by lscheinkman** [comment]
:+1:

**Pull request merged**